doc/spec_TransferElement.dox

```diff
@@ -101,8 +99,11 @@ This document describes the behaviour of the TransferElement base class, the NDR
     - float
     - double
     - std::string
+    - ChimeraTK::Boolean
+    - ChimeraTK::Void
   - 2.2 Applications select a data type for the buffer via the UserType template argument of Device::getZzzRegisterAccessor() resp. DeviceBackend::getRegisterAccessor().
   - 2.3 If needed, the register value is converted between the actual register's data type and the application-selected UserType.
+    - 2.3.1 When the actual type is Void, the conversion is 0 or false in the respective UserType.
```

Comment:
Not only when the actual type is ChimeraTK::Void (I would be very explicit here to distinguish from C++ void). It goes both ways.
If the actual type is void (can the actual type be ChimeraTK::Void, or is it conceptually void?), the buffer is 1x1 of the user type and 0 when reading. When written, the data is discarded.
If the actual type is not void, but the UserType is ChimeraTK::Void, the user buffer has the same shape (number of elements and channels) as the underlying data. When writing, 0 is written to all data elements. Reading is only possible with wait_for_new_data (see 5.2.9)

Reply:
I agree with the explicit distinction 

```suggestion
    - 2.3.1 ChimeraTK::Void is offered as a trigger mechanism rather than data transfer. Therefore the conversion fills the user buffer with 0 or false in the respective UserType for reading with wait_for_new_data (see 5.2.9) and discards any data for writing.
```

Is this too long?

Reply:
a) It's a bit long. We try to have the spec as "what is" to keep it short. There's not so much "why is" in it.
b) It's not quite correct. There are two use  cases which are mixed up in the current wording.

1. The UserType is ChimeraTK::void, but the device has data in it. In this 0 is written to the device, and data coming from the device is discarded. The user Buffer is of type vector<vector<ctk::Void>> and contains elements, but not data content (that's why 0 is "created" when writing and data is discarded when reading into that buffer)
2. The UserType is some data type, but the device conceptually has a void register. When "reading" (i.e. an interrupt arrives, it is seen in the notification queue, user code calls post-read), 0 is filled into the user buffer because we have to invent some data as there is none from the device. When writing (e.g. triggering an action) the data from the UserBuffer is ignored.

Reply:
I would like to keep at least some of the "why". It is always better if a specification is understandable, and not every sentence needs to be a testable condition. So I would keep at least the first sentence, "ChimeraTK::Void is offered as a trigger mechanism rather than data transfer." The second part does mix up the two use cases, but it still captures the correct idea: Conversion to void is achieved by throwing away data, and conversion from void is achieved by filling up the requested shape with zeros. We only need a precise wording. E. g. based on Martins comment,
"If the actual type is void , the user buffer is 1x1 of the UserType. It is 0 after reading. When written, its data is discarded.
If the actual type is not void, but the UserType is ChimeraTK::Void, the user buffer has the same shape (number of elements and channels) as the underlying actual data type. When writing, 0 is written to all data elements. Reading is only possible with wait_for_new_data (see 5.2.9)"